src/app/components/works/work3/work3.component.ts

```diff
@@ -1,15 +1,136 @@
+import { Converter } from '@core/converter';
+import { Gamma } from '@core/gamma';
+import { Avalanche } from '@core/avalanche';
+
 import { Component, OnInit } from '@angular/core';
+import { Label } from 'ng2-charts';
+import { GOST } from '@/app/core/gost';
+import { DES } from '@/app/core/des';
 
 @Component({
   selector: 'app-work3',
   templateUrl: './work3.component.html',
   styleUrls: ['./work3.component.css']
 })
 export class Work3Component implements OnInit {
+  checkedGost = true;
+
+  viewModes = ['Str', 'Bin', 'Hex'];
+  views = {base: 0, gamma: 1, cipher: 0};
+
+  base = 'Hello Darkness my old friend. I\'ve come to talk with you again';
+  gamma = '';
+  cipher = '';
+  vfCase: '0' | '1' | '2' | '3' = '0';
+
+  target: 'base' | 'gamma' = 'base';
+  targetBit = '1';
+
+  researches = '';
+  lineChartData: any[] = [
+    {data: [0], label: 'Количество изменённых бит'},
+  ];
+  lineChartLabels: Label[] = ['0'];
+
+  private setLabels(iterations: 16 | 32 = 16) {
```

Comment:
Fix order of public and private methods according to rules (which are not documented, ha-ha).